src/errors.ts

```diff
@@ -0,0 +1,66 @@
+class VentoError extends Error {
+  constructor(
+    message?: string,
+    public override cause?: Error,
+  ) {
+    super(message);
+    Error?.captureStackTrace(this, this.constructor);
+    this.name = this.constructor.name;
+  }
+}
```

Comment:
What's the purpose of this class?
`TemplateError` and `TransformError` could extend `Error` directly because I don't see here anything that Error doesn't do.

For example, in the other classes, instead of calling `super(message, cause)`,  you can simply call `super(message, { cause })`.

Or am I missing something?

Reply:
Oh, I had this in there so I could dynamically assign this.name, and also call `captureStackTrace`. It was a pattern I borrowed from Eleventy's error handling.

I also came across [this pattern](https://medium.com/@Nelsonalfonso/understanding-custom-errors-in-typescript-a-complete-guide-f47a1df9354c#cdd1) which talks about using `Object.setPrototypeOf`...but I guess this is only applicable when compiling down to ES5...?